addons/sale_timesheet/models/project_update.py

```diff
@@ -26,37 +26,30 @@ def _get_services_values(self, project):
         if not project.allow_billable:
             return {}
         services = []
-        total_sold_per_uom = defaultdict(float)
-        total_effective_per_uom = defaultdict(float)
         total_sold, total_effective, total_remaining = 0, 0, 0
         sols = self._get_project_sols(project)
         for sol in sols:
             #We only want to consider hours and days for this calculation
             if sol.product_uom.category_id == self.env.company.timesheet_encode_uom_id.category_id:
+                name = sol.name_get()[0][1] if len(sols.order_id) > 1 else sol.name
```

Comment:
We could call the name_get before the for loop with all SOLs. The name_get method returns a list of tuple containing the id and the display name. So we could do this before the for loop:µ
```py
name_by_sol = dict(sols.name_get())
```
And then we could do this in the for loop:
```suggestion
                name = name_by_sol[sol.id] if len(sols.order_id) > 1 else sol.name
```